inc/search.class.php

```diff
@@ -6265,7 +6265,7 @@ static function makeTextSearch($val, $not=false) {
             $val = substr($val, $begin, $length-$end-$begin);
          }
 
-         $SEARCH = " $NOT LIKE '".(!$begin?"%":"").$val.(!$end?"%":"")."' ";
+         $SEARCH = " $NOT LIKE '".(!$begin?"%":"").rtrim($val).(!$end?"%":"")."' ";
```

Comment:
Why not just `trim()`? Problem would be the same with an extra space at the begginning of the searched string, isn't it?